### DISCUSSION TITLE
Run at particular time

### DISCUSSION THREAD

**Discussion by sathyarajshetigar** [TomSchimansky/CustomTkinter#1028] [Q&A] [answered]
Hi, Is there an overload method for self.after() where I can run a method at particular time say 1:30PM

**Comment by Akascape** [comment, marked as answer]
@sathyarajshetigar Use a while loop with threading. Use datetime to get the current time and put it in the loop. When it reaches the final time, break the loop and start the new operation.

Here is a good reference you can take: https://www.geeksforgeeks.org/creat-an-alarm-clock-using-tkinter/amp/

**Reply by sathyarajshetigar** [reply]
` now = datetime.datetime.now()
        target_time = datetime.datetime(now.year, now.month, now.day, 15, 24)
        if now > target_time:
            # The target time has passed, schedule the function for tomorrow
            target_time = target_time + datetime.timedelta(days=1)
        # Calculate the number of seconds until the target time
        seconds = (target_time - now).total_seconds()
        # Schedule the function to be called
        self.after(int(seconds * 1000), self.ExitAll)`
        managed to achieve it using the code above. Leaving it here for others.
